### ISSUE TITLE
build fails on darwin

### ISSUE THREAD

**Issue by vangent** [mattn/go-ieproxy#34]
Describe the bug
Building things that use this library seems to yield an error on darwin, both amd64 and arm64.

Example output:

❯ GOOS=darwin GOARCH=arm64 go build .
# github.com/mattn/go-ieproxy
../../Go/pkg/mod/github.com/mattn/go-ieproxy@v0.0.3/ieproxy.go:36:9: undefined: getConf
../../Go/pkg/mod/github.com/mattn/go-ieproxy@v0.0.3/ieproxy.go:41:9: undefined: reloadConf
../../Go/pkg/mod/github.com/mattn/go-ieproxy@v0.0.3/ieproxy.go:48:2: undefined: overrideEnvWithStaticProxy
../../Go/pkg/mod/github.com/mattn/go-ieproxy@v0.0.3/ieproxy.go:53:13: psc.findProxyForURL undefined (type *Proxy

**Comment by mattn** [comment]
Sorry, I don't have macOS. So I can't fix this issue.

**Comment by laurazard** [comment]
On an M1 MacBook Pro on macOS 12.3.1 (21E258), I can't replicate this. Seems to build successfully.

**Comment by vangent** [comment]
https://github.com/google/go-cloud/issues/3114#issuecomment-1093252634

describes how to repro from a user. I think you can do it on any platform by setting 

GOOS=darwin GOARCH=amd64

**Comment by laurazard** [comment]
Oh sorry, I didn't notice this happened building things that _used_ this library. I can replicate, I'll try to take a look myself.

**Comment by laurazard** [comment]
Looks like this happens due to golang/go#24068. I managed to make this work by enabling the `CGO_ENABLED` flag, like so:

`GOOS=darwin GOARCH=amd64 CGO_ENABLED=1 go build`

Could you test this @vangent?

**Comment by caarlos0** [comment]
I could but I don't really want to use CGO...

**Comment by laurazard** [comment]
CGo is used extensively in this package to interface with OS APIs to fetch the relevant configs, so I'm not sure what alternative is left if you're trying to use it. If you want to use parts of the lib that don't rely on C calls, you can fork it and just use the parts you need!

**Comment by caarlos0** [comment]
I understand, I'm getting this as a dependency of a dependency actually, so its not that simple...

on a side note, probably worth mentioning that it uses cgo on the readme/godocs...

**Comment by bep** [comment]
>on a side note, probably worth mentioning that it uses cgo on the readme/godocs...

With big bold letters!

**Comment by chrislusf** [comment]
> CGo is used extensively in this package to interface with OS APIs to fetch the relevant configs, so I'm not sure what alternative is left if you're trying to use it. If you want to use parts of the lib that don't rely on C calls, you can fork it and just use the parts you need!

Not a good option for projects using libraries that is using ieproxy.
```
github.com/Azure/azure-pipeline-go@v0.2.3 github.com/mattn/go-ieproxy@v0.0.1
gocloud.dev@v0.25.0 github.com/mattn/go-ieproxy@v0.0.3
```

**Comment by vangent** [comment]
Note that `gocloud.dev` doesn't use this library directly, it is getting the dependency indirectly from `azure-pipeline-go`.

**Comment by mattn** [comment]
So, would be solved this problem with adding empty implementations for non-cgo?

**Comment by caarlos0** [comment]
I think it might, yes.

Azure could also not use the proxy by default (as I suggested [here](https://github.com/Azure/azure-pipeline-go/issues/31#issuecomment-1095685986)), but having a no-op impl for non-cgo seems even better...

**Comment by chrislusf** [comment]
Got into this when building with 0.0.8

```
# github.com/mattn/go-ieproxy
../../../../../pkg/mod/github.com/mattn/go-ieproxy@v0.0.8/ieproxy.go:53:13: psc.findProxyForURL undefined (type *ProxyScriptConf has no field or method findProxyForURL, but does have FindProxyForURL)
```

**Comment by caarlos0** [comment]
oh, my bad, fix on #43, now with tests!
